src/transformers/models/seaformer/modeling_seaformer.py

```diff
@@ -0,0 +1,764 @@
+            a, b, ks, stride, pad, dilation, groups, bias=bias))
+        
+        bn = nn.BatchNorm2d(b)
+        nn.init.constant_(bn.weight, bn_weight_init)
+        nn.init.constant_(bn.bias, 0)
+        self.add_module('bn', bn)
+        
+    def forward(self, x):
+        return self.bn(self.c(x))
+
+class SeaformerMLP(nn.Module):
+    """
+    Linear Embedding.
+    """
+    def __init__(self, in_features, hidden_features=None, out_features=None, act_layer=nn.ReLU, drop=0.):
+        super().__init__()
+        out_features = out_features or in_features
+        hidden_features = hidden_features or in_features
+        self.fc1 = SeaformerConv2d_BN(in_features, hidden_features)
+        self.dwconv = nn.Conv2d(hidden_features, hidden_features, 3, 1, 1, bias=True, groups=hidden_features)
+        self.act = act_layer()
+        self.fc2 = SeaformerConv2d_BN(hidden_features, out_features)
+        self.drop = nn.Dropout(drop)
+
+    def forward(self, x):
+        x = self.fc1(x)
+        x = self.dwconv(x)
+        x = self.act(x)
+        x = self.drop(x)
+        x = self.fc2(x)
+        x = self.drop(x)
+        return x
+
+class SeaformerInvertedResidual(nn.Module):
+    def __init__(
+            self,
+            inp: int,
+            oup: int,
+            ks: int,
+            stride: int,
+            expand_ratio: int,
+            activations=None
+    ) -> None:
+        super(SeaformerInvertedResidual, self).__init__()
+        self.stride = stride
+        self.expand_ratio = expand_ratio
+        assert stride in [1, 2]
+
+        if activations is None:
+            activations = nn.ReLU
+
+        hidden_dim = int(round(inp * expand_ratio))
+        self.use_res_connect = self.stride == 1 and inp == oup
+
+        layers = []
+        if expand_ratio != 1:
+            # pw
+            layers.append(SeaformerConv2d_BN(inp, hidden_dim, ks=1))
+            layers.append(activations())
+        layers.extend([
+            # dw
+            SeaformerConv2d_BN(hidden_dim, hidden_dim, ks=ks, stride=stride, pad=ks // 2, groups=hidden_dim),
+            activations(),
+            # pw-linear
+            SeaformerConv2d_BN(hidden_dim, oup, ks=1)
+        ])
+        self.conv = nn.Sequential(*layers)
+        self.out_channels = oup
+        self._is_cn = stride > 1
+
+    def forward(self, x):
+        if self.use_res_connect:
+            return x + self.conv(x)
+        else:
+            return self.conv(x)
+
+
+class SeaformerStackedMV2Block(nn.Module):
+    def __init__(
+            self,
+            cfgs,
+            stem,
+            inp_channel=16,
+            activation=nn.ReLU,
+            width_mult=1.):
+        
+        super().__init__()
+        
+        self.stem = stem
+        if stem:
+            self.stem_block = nn.Sequential(
+                SeaformerConv2d_BN(3, inp_channel, 3, 2, 1),
+                activation()
+            )
+            
+        self.cfgs = cfgs
```

Comment:
We generally use the config to access configuration parameters wherever possible, so class initialization arguments such as cfgs and activation should be eliminated ideally